### PR TITLE
ci: Remove pylint

### PR DESCRIPTION
**Pull request by aahung**
### Issue #, if available

Ruff's stability is proven and adopted by many large projects (see [README](https://github.com/charliermarsh/ruff)). So far ruff has been catching a lot of code issues in previous PRs. At by removing pylint we can save a minute or two in CI/CD for each build, each python version.

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by acristin — APPROVED** [review]
:tada:

File: requirements/dev.txt [hunk 1]
```diff
@@ -5,7 +5,6 @@ pytest-cov~=2.10.1
 pytest-xdist~=2.5
 pytest-env~=0.6.2
 pytest-rerunfailures~=9.1.1
-pylint~=2.15.0
 pyyaml~=5.4
 ruff==0.0.237  # loose the requirement once it is more stable
```
Comment (hoffa) [inline requirements/dev.txt:9]: Remove the pin? So always latest in CI.

**Review by xazhao — APPROVED** [review]

**Review by ssenchenko — APPROVED** [review]

File: requirements/dev.txt (hunk 1 above)
Comment (aahung) [reply requirements/dev.txt:9]: why not

Comment (aahung) [reply requirements/dev.txt:9]: it doesn't hurt

File: Makefile [hunk 2]
```diff
@@ -35,13 +35,9 @@ black-check:
 	bin/yaml-format.py --check integration --add-test-metadata
 
 lint:
-	# experimental: use ruff to perform checks before pylint to see
-	# if it can check issues before pylint
 	ruff samtranslator bin schema_source
 	# mypy performs type check
 	mypy --strict samtranslator bin schema_source
-	# Linter performs static analysis to catch latent bugs
-	pylint --rcfile .pylintrc samtranslator schema_source
```
Comment (hoffa) [inline Makefile:44]: So none of that stuff in `.pylintrc` is needed? Is it all enabled by default?

Comment (aahung) [reply Makefile:44]: Check `ruff.toml`

**Review by hoffa — APPROVED** [review]

**Review by GavinZZ — APPROVED** [review]

File: requirements/dev.txt (hunk 1 above)
Comment (aahung) [reply requirements/dev.txt:9]: well it shows its annoyance as 238 introduced more rules.

**Review by ConnorRobertson — APPROVED** [review]

**Comment by charliermarsh** [comment]
Sorry for the drive-by but this is really cool to see and I'm glad Ruff is working for you. Always feel free to file Issues etc. as they come up, I try to be very responsive :)

**Comment by aahung** [comment]
@charliermarsh thanks for working on this amazing tool

**Comment by hoffa** [comment]
@charliermarsh FYI the SAM CLI team is also moving toward Ruff (see e.g. https://github.com/aws/aws-lambda-builders/pull/435). Thanks a lot for working on this, my eyes have been opened

**Pull request merged**